backend/apps/slack/services/message_auto_reply.py

```
@@ -42,35 +55,397 @@ def generate_ai_reply_if_unanswered(message_id: int):
+            return
+
+        try:
+            # One more validation check right before formatting
+            if ai_response_text:
+                response_str = str(ai_response_text).strip()
+                if response_str.upper() in ("YES", "NO"):
+                    logger.error(
+                        "Blocking Question Detector output before formatting",
+                        extra={"channel_id": channel_id, "message_id": message.slack_message_id},
+                    )
+                    raise ValueError("Invalid response: Question Detector output detected")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)